go-controller/pkg/node/gateway_shared_intf.go

```
@@ -2173,3 +2186,86 @@ func generateIPFragmentReassemblyFlow(ofPortPhys string) []string {
+	if err != nil {
+		return fmt.Errorf("failed to list addresses for the link %s: %v", link.Attrs().Name, err)
+	}
+
+	var staleMasqueradeIPs config.MasqueradeIPsConfig
+	for _, addr := range addrs {
+		if addr.Label == fmt.Sprintf("%s-masq", link.Attrs().Name) && !utilnet.IsIPv6(addr.IP) &&
+			!addr.IPNet.Contains(config.Gateway.MasqueradeIPs.V4OVNMasqueradeIP) {
+			if err = config.AllocateV4MasqueradeIPs(iputils.PrevIP(iputils.PrevIP(addr.IP)), &staleMasqueradeIPs); err != nil {
+				return fmt.Errorf("unable to determine stale V4MasqueradeIPs: %s", err)
+			}
+			if err = deleteMasqueradeResources(link, &staleMasqueradeIPs); err != nil {
+				return fmt.Errorf("unable to delete masquerade resources: %s", err)
+			}
+		}
+		if addr.Label == fmt.Sprintf("%s-masq", link.Attrs().Name) && utilnet.IsIPv6(addr.IP) &&
+			!addr.IPNet.Contains(config.Gateway.MasqueradeIPs.V6OVNMasqueradeIP) {
+			if err = config.AllocateV6MasqueradeIPs(iputils.PrevIP(iputils.PrevIP(addr.IP)), &staleMasqueradeIPs); err != nil {
+				return fmt.Errorf("unable to determine stale V6MasqueradeIPs: %s", err)
+			}
+			if err = deleteMasqueradeResources(link, &staleMasqueradeIPs); err != nil {
+				return fmt.Errorf("unable to delete masquerade resources: %s", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// deleteMasqueradeResources() function removes following Linux resources given a
+// config.MasqueradeIPsConfig struct and netlink.Link:
+// - neighbour object for IPv4 and IPv6 OVNMasqueradeIP and DummyNextHopMasqueradeIP.
+// - masquerade route added by addMasqueradeRoute() function while starting up the gateway.
+// - iptables rules created for masquerade subnet based on ipForwarding and Gateway mode.
+func deleteMasqueradeResources(link netlink.Link, staleMasqueradeIPs *config.MasqueradeIPsConfig) error {
+	var subnets []*net.IPNet
+	var neighborIPs []net.IP
+	if config.IPv4Mode {
+		_, masqIPNet, _ := net.ParseCIDR(fmt.Sprintf("%s/32", staleMasqueradeIPs.V4OVNMasqueradeIP.String()))
```

Do not ignore the error please.

---

Ack